lib/pharos/config_schema.rb

```diff
@@ -34,6 +34,16 @@ def self.load(data)
       result = schema.call(DEFAULT_DATA.merge(data))
       raise Pharos::ConfigError, result.messages unless result.success?
 
+      messages = { hosts: {} }
```

Comment:
IMO this can be implemented as predicate

```ruby
    def self.build
      # rubocop:disable Lint/NestedMethodDefinition
      Dry::Validation.Params do
        configure do

          # @param hosts [Array<Hash>]
          def unique_addresses?(hosts)
            return false if hosts.group_by { |host| "#{host[:address]}:#{host[:ssh_port]}" }.size < hosts.size

            true
          end
```

```ruby
required(:hosts).filled(:unique_addresses?, min_size?: 1) do
```